vue-components/src/components/ExportWidget.vue

```diff
@@ -0,0 +1,117 @@
+<script setup lang="ts">
+import { ref, computed, watch } from 'vue'
+
+interface Props {
+  currentDataset: string
+  repositoryDatasets: string[]
+  status: 'idle' | 'pending' | 'success' | 'fail'
+  progress: number
+}
+
+type Events = {
+  exportDataset: [{ name: string; full: boolean }]
+}
+
+const props = defineProps<Props>()
+
+const emit = defineEmits<Events>()
+
+function pathToFilename(path: string): string {
+  return path.split('\\').pop()?.split('/').pop() || path
+}
+
+function filenameToBasename(filename: string) {
+  const splitFilename = filename.split('.')
+
+  if (splitFilename.length <= 1) {
+    return filename
+  } else {
+    return splitFilename.slice(0, -1).join('.')
+  }
+}
+
+const currentName = ref<string>(filenameToBasename(pathToFilename(props.currentDataset)))
+const overwrite = ref<boolean>(false)
+const fullExport = ref<boolean>(false)
+
+const datasetsNames = computed(() => {
+  const datasetNames = props.repositoryDatasets.map(pathToFilename).map(filenameToBasename)
+  return new Set(datasetNames)
+})
+
+const exporting = computed(() => {
+  return props.status === 'pending'
+})
+
+const barColor = computed(() => {
+  switch (props.status) {
+    case 'success':
+      return 'positive'
+    case 'fail':
+      return 'negative'
+    case 'idle':
+    case 'pending':
+    default:
+      return 'primary'
+  }
+})
+
+watch(
+  () => props.currentDataset,
+  (currentDataset) => {
+    currentName.value = filenameToBasename(pathToFilename(currentDataset))
+    overwrite.value = false
+    fullExport.value = false
+  }
+)
+
+function onOverwriteChange(value: boolean, _ev: MouseEvent) {
+  overwrite.value = value
+}
+
+function onNameChange(value: string) {
+  currentName.value = value
+  overwrite.value = false
+}
+
+function onExport() {
+  emit('exportDataset', { name: currentName.value, full: fullExport.value })
+}
+</script>
+
+<template>
+  <q-card-section>
+    <q-input
+      outlined
+      dense
+      stack-label
+      :disable="exporting"
+      :model-value="currentName"
+      :error="datasetsNames.has(currentName) && !overwrite"
```

Comment:
After the first export, the GUI shows the "overwrite" warning.  It makes me thing the first export did not work:
![image](https://github.com/user-attachments/assets/3c4aa434-9145-43da-b0d8-05482af78c05)